src/SFA.DAS.EAS.Web/Orchestrators/EmployerAccountTransactionsOrchestrator.cs

```diff
@@ -314,6 +313,36 @@ public EmployerAccountTransactionsOrchestrator(IMediator mediator, ICurrentDateT
             };
         }
 
+        private static ICollection<TransactionLine> AggregateLevyTransactions(ICollection<TransactionLine> transactions)
+        {
+            if (transactions == null || !transactions.Any())
+                return new List<TransactionLine>();
+
+            if (transactions.Count(t => t.TransactionType == TransactionItemType.Declaration) < 2)
```

Comment:
The use of count will enumerate all items in the collection - this doesn't seem necessary when all we're finding out is whether there is a single item that matches the condition. Whether there is 2 or 2000 doesn't matter.

Reply:
Thinking about it again I wonder if moving the array creation at L325 to the start and then using levytransactions.length < 2 at L321 would make L321 more obvious about what it's doing? Not sure about this - depends on the probability of there being < 2 txns.

Reply:
so I could do this as a do while or even create an extension to do this i guess so that it stops searching after finding x occurrences. I originally just placed the count in there as it was the simplest way to the check for the rest of the code without having to create a new collection.

I'm guessing if there are less that 2 occurrences it would be more efficient to use count as you don't have the overhead of creating a new collection which would still have to enumerate over all of the items in the collection. It would however be more efficient to pre-create levy transactions if there were more than 1 occurrences.

I'll create an extension to take a predicate that can be used to work out if there are x occurrences in a more efficient way.

Reply:
Where(<predicate>).Take(2) then would do it?

Reply:
where would still iterate over the whole collection first.

Reply:
i'll change this now

Reply:
OK after our chat I implemented the linq where.take(2) and then realise the extension I created was even more efficient for the specific task as take does guarantee you'll get two so you have to count them to make sure where as the extension I've added doesn't need to do this. I've at least been refreshed on the chained linq process however.